src/GraphControl/Control/Grapher.cpp

```diff
@@ -118,9 +118,20 @@ namespace GraphControl
     {
         if (m_graph != nullptr && m_renderMain != nullptr)
         {
-            if (auto renderer = m_graph->GetRenderer())
+            if(auto renderer = m_graph->GetRenderer())
             {
-                if (SUCCEEDED(renderer->ResetRange()))
+                HRESULT hr;
+                if (m_resetUsingInitialDisplayRange)
+                {
+                    hr = renderer->SetDisplayRanges(m_initialDisplayRangeXMin, m_initialDisplayRangeXMax, m_initialDisplayRangeYMin, m_initialDisplayRangeYMax);
+                    m_resetUsingInitialDisplayRange = false;
+                }
+                else
+                {
+                    hr = renderer->ResetRange();
```

Comment:
I don't think this gets called in this case:

1. Graph anything
2. Open up graph settings, set max x-range to 3000
3. Move graph around, click reset, resets back to view to when you set x-range to 3000 instead of best fit

Reply:
In Graph Settings we directly call GetDisplayRanges and SetDisplayRanges. In the case where the range was modified via settings, I have added a flag in that case we now replot the graph.